spec/controllers/miq_ae_class_controller_spec.rb

```diff
@@ -406,14 +406,18 @@
       domain1 = FactoryGirl.create(:miq_ae_system_domain_enabled)
 
       domain2 = FactoryGirl.create(:miq_ae_domain_enabled)
+      domain3 = FactoryGirl.create(:miq_ae_git_domain)
       controller.instance_variable_set(:@_params,
-                                       :miq_grid_checks => "aen-#{domain1.id}, aen-#{domain2.id}, aen-someid"
+                                       :miq_grid_checks => "aen-#{domain1.id}, aen-#{domain2.id}, aen-#{domain3.id}, aen-someid"
                                       )
       allow(controller).to receive(:replace_right_cell)
+      expect_any_instance_of(GitBasedDomainImportService).to receive(:destroy_repository).with(domain3.git_repository.id)
```

Comment:
I think this whole test could be cleaned up to better separate setup from execution, and also to avoid `expect_any_instance_of`:

```
let(:domain1) { ... }
let(:domain2) { ... }
let(:domain3) { ... }
let(:git_based_domain_import_service) { double("GitBasedDomainImportService") }

before do
  # all the other setup regarding the controller
  allow(GitBasedDomainImportService).to receive(:new).and_return(git_based_domain_import_service)
end

it "whatever the description is already here is" do
  expect(git_based_domain_import_service).to receive(:destroy_repository).with(domain3.git_repository.id)
  # other expectations. Could also break out the other expectations
  # into separate 'it' blocks, which is probably preferable.
end
```